tox.ini

```diff
@@ -64,6 +64,26 @@ setenv =
     {[base]setenv}
     SYNAPSE_POSTGRES = 1
 
+# A test suite for the oldest supported versions of Python libraries, to catch
+# any uses of APIs not available in them.
+[testenv:py27-old]
+skip_install=True
+deps =
+    # Old automat version for Twisted
+    Automat == 0.3.0
+
+    mock
+    lxml
+commands =
+    /usr/bin/find "{toxinidir}" -name '*.pyc' -delete
+    # Make all greater-thans equals, as well as update psycopg2 and pyopenssl
+    # versions to older versions which still compile on current versions of
```

Comment:
s/older/newer/?

Could you give more specific examples? which OSes do these libs not compile on? (Travis runs on a pretty ancient os (ubuntu 14.04 ?), so I'd be surprised if there were things which we needed to support for xenial which don't compile on there)